app/src/main/java/com/duckduckgo/app/settings/db/SettingsDataStore.kt

```diff
@@ -83,8 +84,16 @@ class SettingsSharedPreferences constructor(private val context: Context, privat
         set(enabled) = preferences.edit { putBoolean(KEY_AUTOCOMPLETE_ENABLED, enabled) }
 
     override var appLoginDetection: Boolean
-        get() = preferences.getBoolean(KEY_LOGIN_DETECTION_ENABLED, true)
-        set(enabled) = preferences.edit { putBoolean(KEY_LOGIN_DETECTION_ENABLED, enabled) }
+        get() = if (variantManager.isFireproofExperimentEnabled()) {
```

Comment:
I take this assumes once the experiment finishes everything goes back to normal and user's choice won't be respected? The discussion is going on in the PR but imo I think we should respect the user's choice here :D

Reply:
I've updated this based on the PR and I also removed the additional `KEY_AUTO_LOGIN_DETECTION_ENABLED` pref (It isn't required now that we're persisting the choice from the experiment).